store/src/java-test/com/zimbra/cs/mailbox/MailboxTest.java

```diff
@@ -640,6 +640,63 @@ public void testGetModifiedItemsCount() throws Exception {
         Assert.assertEquals(2, count);
     }
 
+    @Test
+    public void testAdditionalQuotaProviderExceedsQuota() throws Exception {
+        AdditionalQuotaProvider additionalQuotaProvider = new AdditionalQuotaProvider() {
+            @Override
+            public long getAdditionalQuota(Mailbox mailbox) {
+                return 10;
+            }
+        };
+        MailboxManager.getInstance().addAdditionalQuotaProvider(additionalQuotaProvider);
+        Provisioning prov = Provisioning.getInstance();
+        Map<String, Object> attrs = new HashMap<String, Object>();
+        attrs.put("zimbraMailQuota", "5");
+        Account acct = prov.createAccount("testAdditionalQuotaProvider@zimbra.com", "secret", attrs);
+        Mailbox mbox = MailboxManager.getInstance().getMailboxByAccount(acct);
+        try {
+            mbox.checkSizeChange(0);
+            Assert.fail("Expected QUOTA_EXCEEDED exception");
+        }
+        catch (MailServiceException ignored) {}
+
+        Assert.assertEquals(10L, mbox.getSize());
```

Comment:
Shouldn't the mbox size be 15(zimbraMailQuota + AdditionalQuota)?

Reply:
zimbraMailQuota is the account size limit (5 in this test). mbox.getSize() returns total occupied space that is 0 (mailbox is empty) plus 10 (quota provided by the additional provider). Hence the check needs to be against 10.